### ISSUE TITLE
Update to 6.4.2

### ISSUE THREAD

**Issue by m-pire** [directus/directus#1768]
# Version Info
  - Directus version and branch: v6.4.2 (Ruby Rabbit)
  - PHP version: 7.0.19
  - MySQL version: MariaDB 10.1.23
  - Web server: Apache 2.2
  - OS name and version: Debian 9
  - Directus 6.3.9 is running well without any problems

## Expected Behavior
Update from Directus 6.3.9 (with PHP SDK 1.0) to 6.4.2 (with PHP SDK 2.0)

## Actual Behavior
Update not possible - I think that there is dependency problem with composer.json, which I cant solve.

## Steps to Reproduce
 1. Clean new installation of Directus 6.4.2
 2. Configuring api/config.php and api/configuration.php to existing Database
 3. Running DB Migration Script
 4. Adding "directus/sdk": "0.2.*" to composer.json
 4. Composer install & update

OR

1. Unzip Directus 6.4.2 over Directus 6.3.9
2. Running DB Migration Script
3. Adding "directus/sdk": "0.2.*" to composer.json
4. Composer install & update

## Errors in Frontend using PHP SDK 2:

object(PDOException)#149 (8) {
  ["message":protected]=>
  string(84) "SQLSTATE[42S22]: Column not found: 1054 Unknown column 'hidden_list' in 'field list'"
  ["string":"Exception":private]=>
  string(0) ""
  ["code":protected]=>
  string(5) "42S22"
  ["file":protected]=>
  string(111) "/var/www/users/webs/tawl/webroot/directus_new/vendor/zendframework/zend-db/src/Adapter/Driver/Pdo/Statement.php"
  ["line":protected]=>
  int(239)
  ["trace":"Exception":private]=>
  array(12) {
    [0]=>
    array(6) {
      ["file"]=>
      string(111) "/var/www/users/webs/tawl/webroot/directus_new/vendor/zendframework/zend-db/src/Adapter/Driver/Pdo/Statement.php"
      ["line"]=>
      int(239)
      ["function"]=>
      string(7) "execute"
      ["class"]=>
      string(12) "PDOStatement"
      ["type"]=>
      string(2) "->"
      ["args"]=>
      array(0) {
      }
    }
    [1]=>
    array(6) {
      ["file"]=>
      string(102) "/var/www/users/webs/tawl/webroot/directus_new/vendor/directus/database/Schemas/Sources/MySQLSchema.php"
      ["line"]=>
      int(322)
      ["function"]=>
      string(7) "execute"
      ["class"]=>
      string(36) "Zend\Db\Adapter\Driver\Pdo\Statement"
      ["type"]=>
      string(2) "->"
      ["args"]=>
      array(0) {
      }
    }

## Errors in Directus: 
Login - "Server Error occured"

## Errors on directus/api
Fatal error: Uncaught Error: Call to undefined method Directus\Permissions\Acl::isPublic() in /var/www/users/webs/tawl/webroot/directus_new/api/core/Directus/Bootstrap.php:692

-> But "Permissions" is installed:
  [...]
  - Installing directus/permissions (dev-version/6.4 da562f2) Cloning da562f2645 from cache
  - Installing directus/sdk (0.2.0) Downloading: 100%
  [...]

## API-Log:

ERROR - 2017-08-26T12:53:22+02:00 - TypeError: Argument 1 passed to Directus\Database\TableGateway\DirectusSettingsTableGateway::__construct() must implement interface Zend\Db\Adapter\AdapterInterface, null given, called in /var/www/users/webs/tawl/webroot/directus_new/vendor/directus/database/TableGatewayFactory.php on line 67 and defined in /var/www/users/webs/tawl/webroot/directus_new/vendor/directus/database/TableGateway/DirectusSettingsTableGateway.php:21
Stack trace:
#0 /var/www/users/webs/tawl/webroot/directus_new/vendor/directus/database/TableGatewayFactory.php(67): Directus\Database\TableGateway\DirectusSettingsTableGateway->__construct(NULL, Object(Directus\Permissions\Acl))
#1 /var/www/users/webs/tawl/webroot/directus_new/api/core/functions.php(1307): Directus\Database\TableGatewayFactory::create('directus_settin...')
#2 /var/www/users/webs/tawl/webroot/directus_new/api/api.php(520): get_project_info()
#3 [internal function]: {closure}()
#4 /var/www/users/webs/tawl/webroot/directus_new/vendor/slim/slim/Slim/Slim.php(598): call_user_func(Object(Closure))
#5 /var/www/users/webs/tawl/webroot/directus_new/vendor/slim/slim/Slim/Slim.php(1367): Slim\Slim->notFound()
#6 /var/www/users/webs/tawl/webroot/directus_new/vendor/slim/slim/Slim/Middleware/Flash.php(85): Slim\Slim->call()
#7 /var/www/users/webs/tawl/webroot/directus_new/vendor/slim/slim/Slim/Middleware/MethodOverride.php(92): Slim\Middleware\Flash->call()
#8 /var/www/users/webs/tawl/webroot/directus_new/api/core/Directus/Slim/CorsMiddleware.php(24): Slim\Middleware\MethodOverride->call()
#9 /var/www/users/webs/tawl/webroot/directus_new/vendor/slim/slim/Slim/Slim.php(1302): Directus\Slim\CorsMiddleware->call()
#10 /var/www/users/webs/tawl/webroot/directus_new/api/core/Directus/Application/Application.php(103): Slim\Slim->run()
#11 /var/www/users/webs/tawl/webroot/directus_new/api/api.php(533): Directus\Application\Application->run()
#12 {main}

ERROR - 2017-08-26T12:53:22+02:00 - ErrorException: Uncaught Error: Call to undefined method Directus\Permissions\Acl::isPublic() in /var/www/users/webs/tawl/webroot/directus_new/api/core/Directus/Bootstrap.php:692
Stack trace:
#0 [internal function]: Directus\Bootstrap::Directus\{closure}(Object(Directus\Hook\Payload))
#1 /var/www/users/webs/tawl/webroot/directus_new/vendor/directus/hook/Emitter.php(266): call_user_func_array(Object(Closure), Array)
#2 /var/www/users/webs/tawl/webroot/directus_new/vendor/directus/hook/Emitter.php(136): Directus\Hook\Emitter->executeListeners(Array, Array, 1)
#3 /var/www/users/webs/tawl/webroot/directus_new/api/core/Directus/Application/Application.php(156): Directus\Hook\Emitter->apply('response', Object(Directus\Hook\Payload))
#4 /var/www/users/webs/tawl/webroot/directus_new/api/core/Directus/Application/Application.php(222): Directus\Application\Application->triggerFilter('response', Object(Directus\Hook\Payload))
#5 /var/www/users/webs/tawl/webroot/directus_new/api/core/Directus/Application/Application.php(137): Directus\A in /var/www/users/webs/tawl/webroot/directus_new/api/core/Directus/Bootstrap.php:692
Stack trace:
#0 [internal function]: Directus\Exception\ExceptionHandler->handleShutdown()

## Error in installation Wizard:
"Try installing dependencies using composer install"

**Comment by WellingGuzman** [comment]
Hey @m-pire, thanks for the report, but why are you trying to add the SDK to Directus?

What are you trying to accomplish? so far the SDK was meant to be used outside the Directus repo, most code from the SDK already exits on Directus code.

**Comment by m-pire** [comment]
Hey so far ;) Currently we are running Directus with the PHP SDK without problems on the same server - is there no need to do so? 

In addition: Do you have any thoughts for the backend problem?

**Comment by WellingGuzman** [comment]
When does error were happening?

Also how are you using the SDK within Directus?

**Comment by m-pire** [comment]
Hi @WellingGuzman,

thank you for your response. I am using this code, as explained in the SDK docs:

`$this->directus = \Directus\SDK\ClientFactory::create(array('database' => array('host' => DB_HOST, 'username' => DB_USER, 'password' => DB_PASSWORD, 'database' => DB_NAME, 'charset' => DB_CHARSET), 'filesystem' => array('root' => $global_config['directus']['root_url'])));`

All requests are going to "$this->directus" - is there a better way to access directus in version 6.4? If you understand you, it means i dont need the PHP SDK anymore?

The error happens here:
`    $articles = $this->directus->getItems('Articles', $params = [
      'status' => '1'
    ]);
`

**Comment by WellingGuzman** [comment]
What I meant is, what the use you are giving the SDK within directus?

You need to SDK if you are developing outside, although the SDK it's a easier way to interact with.

Did you update the SDK/Directus vendor?

**Comment by m-pire** [comment]
Hi, I am sorry, i don't understand your question. I have a project that's running this way

- webroot
-- directus
-- project

So, what is the easiest way to acces the directus data by $client->getItem within PHP?

Thank you in advance.

**Comment by WellingGuzman** [comment]
you should add the directus sdk to the `project` directory, outside the directus vendors.

**Comment by m-pire** [comment]
> The Version 0.2.0 (https://github.com/directus/directus-sdk-php/releases/tag/0.2.0) for 6.4 beta?

Does this solve the issue 
Unknown column 'hidden_list' in 'field list'"
?

**Comment by benhaynes** [comment]
@WellingGuzman – any thoughts?

**Comment by WellingGuzman** [comment]
@m-pire yes, it should fix that missing `hidden_list` field column was removed in 6.4

**Comment by m-pire** [comment]
Hi, I am not sure why this could solve this - which version of the sdk should I use?

**Comment by WellingGuzman** [comment]
If you are using directus `6.3.x` you should use the php sdk `0.1.x` and if you are using Directus 6.4, you should use php sdk `0.2.x`.

**Comment by m-pire** [comment]
Hello again,
I think I can add new information here. When I do as you told me, I got the same error again. 
Please remember, I followed the update instructions on the docs site.
After doing:

ALTER TABLE directus_columns ADD list_view varchar(200) DEFAULT NULL
ALTER TABLE directus_columns ADD hidden_list tinyint(1) NOT NULL DEFAULT '0'

and same for "directus_tables", I got this error:
"Key app.settings is not defined."

When I downgrade the SDK to 0.1.* via composer - it works - the backend and the frontend.

But I see no way to understand why this happens and whats wrong with the schema migration script (php bin/directus db:upgrade) ? 
So I cant upgrade my Installation to Directus 6.4 with the new SDK 0.2.*, sadly. 

I would be very glad if you can look into this. I am so frustrated, trying to upgrade almost 10 times yes.

**Comment by WellingGuzman** [comment]
Where did you get the code below from?

```
ALTER TABLE directus_columns ADD list_view varchar(200) DEFAULT NULL
ALTER TABLE directus_columns ADD hidden_list tinyint(1) NOT NULL DEFAULT '0'
```

Can you show us some screenshot and logs, I'm not following this issue anymore, are you still using the SDK within Directus composer? because it should be out of the Directus `composer.json` itself.

You probably can't use it because those uses the same packages and probably the code from the SDK is older, and they both use the same code, that's why on 6.3, there wasn't any problem, but you should add the SDK to the `composer.json`, you probably can, and we can probably squeeze it, but it's not recommended yet.

can you share your `composer.json` file too?

**Comment by m-pire** [comment]
Hi @WellingGuzman,

> ALTER TABLE directus_columns ADD list_view varchar(200) DEFAULT NULL
> ALTER TABLE directus_columns ADD hidden_list tinyint(1) NOT NULL DEFAULT '0'

I wrote the code on my own to get the SDK working. 

> Can you show us some screenshot and logs

**The error I get:**
Column not found: 1054 Unknown column 'DT.list_view' in 'field list'

> are you still using the SDK within Directus composer?

No, as you told me, I put them now outside the directus folder.

This is my `composer.json` within `directus` folder:
{
  "require": {
    "php": ">=5.6.0",
    "slim/slim": "2.6.2",
    "slim/extras": "dev-master",
    "twig/twig":  "1.*",
    "zendframework/zend-db": "dev-directus",
    "directus/migrations": "dev-master",
    "league/flysystem": "^1.0",
    "swiftmailer/swiftmailer": "^5.4",
    "league/oauth2-google": "^2.0",
    "league/oauth2-github": "^2.0",
    "league/oauth2-facebook": "^2.0",
    "league/oauth1-client": "^1.7"
  },
  "require-dev": {
    "phpunit/phpunit": "~3.7.0"
  },
  "suggest": {
    "paragonie/random_compat": "Generates cryptographically more secure pseudo-random bytes"
  },
  "_require": {
    "rackspace/php-opencloud": "dev-master",
    "amazonwebservices/aws-sdk-for-php": "1.6.2",
    "aws/aws-sdk-php": "2.6.*@dev"
  },
  "autoload": {
    "psr-4": {
      "Directus\\": "api/core/Directus/",
      "Directus\\API\\Routes\\": "api/routes/",
      "Directus\\Installation\\": "installation/includes/",
      "Directus\\Customs\\Embed\\Provider\\": "customs/embeds/",
      "Directus\\Customs\\Hooks\\": "customs/hooks/",
      "Directus\\Customs\\Hasher\\": "customs/hashers/"
    },
    "files": [
      "api/core/functions.php"
    ]
  },
  "minimum-stability": "dev",
  "repositories": [
    {
      "type": "git",
      "url": "https://github.com/WellingGuzman/Slim-Extras.git"
    },
    {
      "type": "git",
      "url": "https://github.com/wellingguzman/zend-db"
    }
  ]
}

**-> I guess this is default. And this ist my `composer.json` within the project folder:**

{
  "require": {
    "directus/sdk": "0.1.*",
    "twig/twig": "^2.0"
  },
  "minimum-stability": "dev",
  "repositories": [
    {
      "type": "git",
      "url": "https://github.com/wellingguzman/zend-db"
    }
  ]
}

**Comment by m-pire** [comment]
Some news! I Tried to setup a very clean and new installation of Directus 6.4.3 and PHP-SDK 0.2.0.
What I did:

1. Downloading the latest release build from [here](https://github.com/directus/directus/releases)
2. Running the installation wizard and checked that Directus is running well with test data
3. Installing Composer in my project folder
4. Running composer install with this composer.json:

{
  "require": {
    "directus/sdk": "0.2.*"
  },
  "minimum-stability": "dev",
  "repositories": [
    {
      "type": "git",
      "url": "https://github.com/wellingguzman/zend-db"
    }
  ]
}

 5. Got exactly the same error:

> Column not found: 1054 Unknown column 'hidden_list' in 'field list'


6. What I am wrong? **OR:** Is there any other way to access the Directus database by PHP **without** use of the SDK?

**Comment by benhaynes** [comment]
Hey @m-pire – you can access Directus content with the API (instead of the SDK). If you're trying to use the SDK, that's not a problem, but I think you're structuring things the wrong way. For example:

*What you're doing*
1. Installing Directus and trying to add the SDK INTO the same composer file.

*What you SHOULD do*
1. Install Directus
2. Create a separate project (your "app") and install the Directus SDK there

Basically your app (SDK) should be in its own directory, with its own composer file. You can't combine these two things together as you'll get many issues. For example, you can install Directus in  a directory – and then something like this in a separate directory:

https://github.com/directus/webapp-example

Hope this helps!

**Comment by m-pire** [comment]
Hi Ben, 

I don't use the same composer.json anymore, I already divided it and set up a whole new installation the way you just told me. With the same error. 

I think there is an error within the database access of the SDK.

For sure, the API could be a solution - is there any way to use this without having GET requests from my servers project to my servers directus installation? It's incredibly slow so far ...

Thanks in advance!

**Comment by WellingGuzman** [comment]
Hey @m-pire, I'm sorry, I'm more confused now.

When you are getting the error:

- Which Directus version are you using?
- Which SDK version are you using?
- Are you connecting to the API or directly to the database?
- What's the error you are getting? Only the missing `hidden_list`?

If you are getting `hidden_list` you should run `bin/directus db:upgrade`.

Why can you upgrade using `bin/directus db:upgrade`? any error?

**Comment by m-pire** [comment]
Hey @WellingGuzman, 

- Which Directus version are you using: **6.4.2**
- Which SDK version are you using: **0.2.0**
- Are you connecting to the API or directly to the database: **Database connection** 
- What's the error you are getting? Only the missing hidden_list?

`Column not found: 1054 Unknown column 'hidden_list' in 'field list'"`
`Column not found: 1054 Unknown column 'list_view' in 'field list'"`

> If you are getting hidden_list you should run bin/directus db:upgrade.
> Why can you upgrade using bin/directus db:upgrade? any error?

I already done this (like upgrade info in the docs):
`db:setup - table 'directus_schema_migrations' already exists. Nothing to do.`
`db:migrate - Migration successful`

**Comment by WellingGuzman** [comment]
Hey @m-pire, I've updated the PHP SDK version to 1.0.0, which will fix the `hidden_list` issue.

Please let us know.

**Comment by m-pire** [comment]
Hey @WellingGuzman,

youre great, this fixes the issue completely. Thank you very much for your time and patience! 
One last question: Is there any use case for PHP [SDK 0.2.0](https://github.com/directus/directus-sdk-php/releases/tag/0.2.0)?

**Comment by WellingGuzman** [comment]
I just made a mistake on tagging the version `0.2.0`, you can freely update to `1.0.0`, which is `0.2.0` with the fixes.

I will eventually remove it, to avoid confusion.

**Comment by WellingGuzman** [comment]
`0.x` is for 6.3 (except 0.2.x) and `1.x` is for 6.4.

Confusingly weird.

**Comment by m-pire** [comment]
Maybe this could be mentioned on the download page of the SDK's ;)

Thanks again!

**Comment by WellingGuzman** [comment]
I added notes about 1.0.0 supports. 6.4 and 0.1, supports 6.3 etc
